_rules/aria-alertdialog-input-error-6f484a.md

```diff
@@ -0,0 +1,915 @@
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
+
+The rule applies to each HTML or SVG element:
+
+- that has one of the following [semantic roles](#semantic-role): `checkbox`, `combobox`, `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`;
+- for which [input errors](https://www.w3.org/TR/WCAG21/#dfn-input-error) are [automatically detected](#automatic-error-detection).
+
+**Note**: The list of applicable [semantic roles](#semantic-role) is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+## Expectation
+
+After [user completion](#completed-input-field) of the target element or triggering the submission of the form if the target element belongs to one, each target element for which an [input error](https://www.w3.org/TR/WCAG21/#dfn-input-error) has been [automatically detected](#automatic-error-detection) causes an element with `role="alertdialog"` to become [visible](#visible), where the following is true:
+
+- it has at least one [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) that is a [focusable](#focusable) element; and
+- the focus moves to the [focusable](#focusable) element; and
+- it is not possible to move the focus away from elements that are [descendents](https://dom.spec.whatwg.org/#concept-tree-descendant) of the element with `role="alertdialog"` until it no longer is [visible](#visible); and
+- after the element with `role="alertdialog"` is no longer [visible](#visible) the focus returns to the position where it was before the element with `role="alertdialog"` became [visible](#visible); and
+- the element with `role="alertdialog"` has an [accessible name](#accessible-name) that is not empty (`""`); and
+- the text content of the element with `role="alertdialog"` identifies the [input error](https://www.w3.org/TR/WCAG21/#dfn-input-error).
+
+**Note**: Information to identify an input error must include information about the element or elements in which the error occurred, and assist the user in understanding what was the cause of the error.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [ARIA18: Using aria-alertdialog to Identify Errors](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA18)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+alertdialog triggered on form submission presents message identifying the input error.
+
+```html
+<html>
+	<head>
+		<meta charset="UTF-8" />
+		<title>Using aria-alertdialog to Identify Errors</title>
+		<script src="http://code.jquery.com/jquery.js"></script>
```

Comment:
I'm not too keen to use jQuery for this. Firstly because it requires hitting the network to run these examples, but also because it create an unnecessary dependency. You can write all of this stuff in native DOM. querySelector solved a lot of problems.